addons/project/static/src/project_sharing/project_sharing.js

```diff
@@ -0,0 +1,75 @@
+/** @odoo-module **/
+
+import { registry } from '@web/core/registry';
+import { useBus, useEffect, useService } from '@web/core/utils/hooks';
+import { ActionContainer } from '@web/webclient/actions/action_container';
+import { MainComponentsContainer } from "@web/core/main_components_container";
+import { useOwnDebugContext } from "@web/core/debug/debug_context";
+import { DebugMenu } from "@web/core/debug/debug_menu";
+import { ErrorHandler, NotUpdatable } from "@web/core/utils/components";
+import { session } from '@web/session';
+
+const { Component } = owl;
+
+export class ProjectSharingWebClient extends Component {
+    setup() {
+        window.parent.document.body.style.margin = "0"; // remove the margin in the parent body
+        this.actionService = useService('action');
+        this.user = useService("user");
+        useService("legacy_service_provider");
+        useOwnDebugContext({ categories: ["default"] });
+        if (this.env.debug) {
+            registry.category("systray").add(
+                "web.debug_mode_menu",
+                {
+                    Component: DebugMenu,
+                },
+                { sequence: 100 }
+            );
+        }
+        useBus(this.env.bus, "ACTION_MANAGER:UI-UPDATED", (mode) => {
+            if (mode !== "new") {
+                this.el.classList.toggle("o_fullscreen", mode === "fullscreen");
+            }
+        });
+        useEffect(
+            () => {
+                this._showView();
+            },
+            () => []
+        );
+    }
+
+    mounted() { }
+
+    handleComponentError(error, C) {
+        // remove the faulty component
+        this.Components.splice(this.Components.indexOf(C), 1);
+        /**
+         * we rethrow the error to notify the user something bad happened.
+         * We do it after a tick to make sure owl can properly finish its
+         * rendering
+         */
+        Promise.resolve().then(() => {
+            throw error;
+        });
+    }
+
+    async _showView() {
+        const { action_name, project_id, can_edit } = session;
+        await this.actionService.doAction(
```

Comment:
Yeah, at the beginning we just added the kanban and form view. But we also add the list view since it is the default view in the classic portal views. So, the portal user has the both views to see the tasks list in the shared project. 🙂